src/libraries/Microsoft.Extensions.Logging.EventSource/src/LoggingEventSource.cs

```diff
@@ -128,50 +128,81 @@ private LoggingEventSource() : base(EventSourceSettings.EtwSelfDescribingEventFo
         /// FormattedMessage() is called when ILogger.Log() is called. and the FormattedMessage keyword is active
         /// This only gives you the human readable formatted message.
         /// </summary>
-        [Event(1, Keywords = Keywords.FormattedMessage, Level = EventLevel.LogAlways)]
+        [Event(1, Keywords = Keywords.FormattedMessage, Level = EventLevel.LogAlways, Version = 2)]
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
             Justification = WriteEventCoreSuppressionJustification)]
-        internal unsafe void FormattedMessage(LogLevel Level, int FactoryID, string LoggerName, int EventId, string? EventName, string FormattedMessage)
+        internal unsafe void FormattedMessage(
+            LogLevel Level,
+            int FactoryID,
+            string LoggerName,
+            int EventId,
+            string? EventName,
+            string? FormattedMessage,
+            string? ActivityTraceId,
+            string? ActivitySpanId,
+            string? ActivityTraceFlags)
         {
-            if (IsEnabled())
+            Debug.Assert(LoggerName != null);
+
+            EventName ??= string.Empty;
+            FormattedMessage ??= string.Empty;
+            ActivityTraceId ??= string.Empty;
```

Comment:
Should we just pass Empty instead of passing null and then recheck? this apply to other added activity properties.

Reply:
Updated